### DISCUSSION TITLE
utils.py or the config.ini parameters

### DISCUSSION THREAD

**Discussion by nighteagle1974** [Louisvdw/dbus-serialbattery#637] [Q&A]
Hello, i new in this driver - and so far  it is working now on my Cerbo GX with an DALY BMS 200A 8S on an CATL 8S LIFEPO4 271Ah Battery.

Now i study the parameters and i compare to the the datasheet of the CATL CELLS and also the BMS.
In the Datasheet of the CATL-CELLS it say absolute minium cell-voltage is 2.5V - and behind they wrote 2.0 V - but i think 2.5 V is deep enough.  This is my BMS Cut-OFF Voltage. The maximum voltage of the cells are 3.65 V also the Cut-OFF Voltage for BMS.

So now what i set in the config / utils.py?

```python
MIN_CELL_VOLTAGE = 2.9
MAX_CELL_VOLTAGE = 3.45
FLOAT_CELL_VOLTAGE = 3.35
MAX_VOLTAGE_TIME_SEC = 15*60
SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT = 90_
```

Charging over 1500W PV it means with the two Victron MPPT 100/50 around 60A in maximum - theoreticaly... then Loading from Power-Supply max 16A. I think 80A Chargecurrent here as parameter? Then i have 230V AC Inverter with needs 125A and some other things that need around 50A. I set Dischargecurrrent to 180A

```python
# battery Current limits
MAX_BATTERY_CHARGE_CURRENT = 80.0
MAX_BATTERY_DISCHARGE_CURRENT = 180_
```

Capacity is 271Ah

_# Daly settings
Battery capacity (amps) if the BMS does not support reading it 
BATTERY_CAPACITY = 271_
```

But i confused about this:

```python
#charge current limits
CC_CURRENT_LIMIT1 = 5
CC_CURRENT_LIMIT2 = MAX_BATTERY_CHARGE_CURRENT/4
CC_CURRENT_LIMIT3 = MAX_BATTERY_CHARGE_CURRENT/2

#discharge current limits
DC_CURRENT_LIMIT1 = 5
DC_CURRENT_LIMIT2 = MAX_BATTERY_DISCHARGE_CURRENT/4
DC_CURRENT_LIMIT3 = MAX_BATTERY_DISCHARGE_CURRENT/2_
```

What this parameter mean?

And the important thing - why the utils.py is overwritten with the defaults after reboot? I have change all parameters with nano - save and exit and then go again in utils.py and all is fine. After reboot utils.py shows default settings... ?

**Comment by mr-manuel** [comment]
> MIN_CELL_VOLTAGE = 2.9
> MAX_CELL_VOLTAGE = 3.45
> FLOAT_CELL_VOLTAGE = 3.35
> MAX_VOLTAGE_TIME_SEC = 15*60
> SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT = 90

Leave it as it is. This are the recommendations.

> # battery Current limits
> MAX_BATTERY_CHARGE_CURRENT = 80.0
> MAX_BATTERY_DISCHARGE_CURRENT = 180

I would reccomend you to set this limits to a bit under the maximum value that your BMS are capable of. If your BMS allows 200 A, then I would set it to 190 A.

> #charge current limits
> #discharge current limits

Don't  change that.

How did you install the driver? Probably with a USB stick? If yes, then please remove the USB stick else it will override your data on every reboot. If not in which path did you edit the `utils.py`?

**Reply by nighteagle1974** [reply]
Thanks for your quick response!

So you wrote don't change this:
#charge current limits
#discharge current limits

What does it mean?
Because on the Cerbo i see this - 5A and 60A - and i mean is not correct?

![IMG_0201](https://github.com/Louisvdw/dbus-serialbattery/assets/1252182/8244a607-89b7-4855-a44d-ca9bae52f3de)

**Reply by mr-manuel** [reply]
Pleaase install https://github.com/Louisvdw/dbus-serialbattery/releases/tag/v1.0.20230508beta it will make things a lot easier.

**Reply by nighteagle1974** [reply]
So i use the /tmp/install.sh script - and the wiki shows me option 1 and 2 but it is not the same as now. 1 Latest release  and 4 is specific version. So here correct the wiki again?

**Reply by nighteagle1974** [reply]
So i have changed the config.ini with few settings like Chargecurrent = 80A and Disscharge current 190A and Capacity=271 and BMS Type = Daly - by the way they are some double names on the comments - hope there load both Daly-Types?

Because some things are not correct? See images... Capacity is now on 30A ? And the Chargecurrent is wrong... some weird value 15,20A?

![IMG_0204](https://github.com/Louisvdw/dbus-serialbattery/assets/1252182/7f58cb5e-538f-423a-ba1b-28361bda6220)
![IMG_0203](https://github.com/Louisvdw/dbus-serialbattery/assets/1252182/dc71bc40-35e1-4221-9787-7f8ab171a1e1)

**Reply by nighteagle1974** [reply]
Here the config.ini

```ini
[DEFAULT]

; If you want to add custom settings, then check the settings you want to change in "config.default.ini"
; and add them below to persist future driver updates.

; --------- Additional settings ---------
; Specify only one BMS type to load else leave empty to try to load all availabe
; LltJbd, Ant, Daly, Daly, Jkbms, Lifepower, Renogy, Renogy, Ecs
BMS_TYPE =Daly


; --------- Battery Current limits ---------
MAX_BATTERY_CHARGE_CURRENT    = 80.0
MAX_BATTERY_DISCHARGE_CURRENT = 190.0

; Battery capacity (amps) if the BMS does not support reading it
BATTERY_CAPACITY = 271
```

**Reply by mr-manuel** [reply]
> So i use the /tmp/install.sh script - and the wiki shows me option 1 and 2 but it is not the same as now. 1 Latest release and 4 is specific version. So here correct the wiki again?

The open pull request was not merged yet...

**Reply by mr-manuel** [reply]
For the capacity check your Daly BMS settings. Probably you have set there 30 Ah, since the value from the config is only fetched, if it's not able to read from BMS. As the description says.

In the second picture you see, that you have the linear mode enabled and that the CCL is limited by SoC.

https://github.com/Louisvdw/dbus-serialbattery/blob/88e5eea93331ae99dc28fcf8b5a40cb866e432e2/etc/dbus-serialbattery/config.default.ini#L119-L146

**Reply by mr-manuel** [reply]
What do you mean with "there are some double names in the comments"?

**Reply by nighteagle1974** [reply]
On the Daly BMS you can not set the Capacity - only Charge and Dischargecurrent and the Cell-Voltages. I have set all this with bluetooth before i use the victron thing. By the way in the other firmware >0,14.3 it is correct 271Ah on the Victron GUI -  like i set in the utils.py. But now where the 271Ah i set in the config.ini with this 1.XX Version?

Image from DALY APP Settings

![Screenshot_20230512-105659_SMART BMS](https://github.com/Louisvdw/dbus-serialbattery/assets/1252182/48e68c5e-d496-43d6-8ff5-815260a93623)

**Reply by nighteagle1974** [reply]
Ah... have found the problem.... i'm an idiot... have not see the menues from the daly right slide... yes i can set the capacity...

**Comment by nighteagle1974** [comment]
Thank you for answer...

Utils.py problem:

Have open an issue - have a look. Was the SD-Card inside Cerbo GX and i forget it because have installed over SSH like wiki said.

**Reply by mr-manuel** [reply]
I corrected that.
